### PR TITLE
Update Pattern page header image

### PR DESCRIPTION
**Pull request by matt-west**
## Proposed Changes

* Update the header background image to a PNG to avoid issues with the pattern tiles appearing pixelated.
* Fix alignment so that the right side of the image aligns with the columns below.

## Testing Instructions

* Run the app.
* Visit http://calypso.localhost:3000/patterns.
* Observe that the background image no longer has white borders or is pixelated.

### Before
Screenshot 2024-03-05 at 13 41 30

### After
Screenshot 2024-03-05 at 13 41 04

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by nightnei — APPROVED** [review]
LGTM 👍

File: client/my-sites/patterns/components/header/style.scss [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 	.patterns-header__inner {
 		padding: 105px 0 64px 84px;
-		background: url(./images/background.svg) no-repeat right bottom;
+		background: url(./images/background@3x.png) no-repeat bottom right 84px / auto 100%;
```
Comment (fredrikekelund) [inline client/my-sites/patterns/components/header/style.scss]:
```suggestion
		background-image: url(./images/background@3x.png);
		background-repeat: no-repeat;
		background-position: bottom right 84px;
		background-size: auto 100%;
```
Not a blocker, but I advocate using the non-shorthand rules in the source code for improved readability. Especially in cases like this where we specify so many rules.

**Review by fredrikekelund — APPROVED** [review]
LGTM 👍 I left one suggestion, but that's not a blocker.

File: client/my-sites/patterns/components/header/style.scss (hunk 1 above)
Comment (matt-west) [reply client/my-sites/patterns/components/header/style.scss]: 🙌 I agree!

I continued with the shorthand here as I thought that was a convention we used. I much prefer the individual rules though.

**Pull request merged**